README.md

```diff
@@ -33,3 +33,83 @@ Your pull requests are welcome but please keep them manageable and focussed.
 Equally your input on any pull requests we have in flight at any given time is invaluable to us, so please get involved.
 
 Thanks! :grin:
+
+
+## Running the example
+
+* Clone the repo
+* cd to `example` folder
+* run `flutter packages get` to install dependencies
+* `flutter run` will start the application on connected android / iOS device
+
+
+## Using the Realtime API
+
+##### Import the package
+```dart
+import 'package:ably_flutter_plugin/ably.dart' as ably;
+```
+
+##### Create an Ably instance
+```dart
+final ably.Ably ablyPlugin = ably.Ably();
+```
+
+##### create a ClientOptions
+```dart
+final clientOptions = ably.ClientOptions.fromKey("<YOUR APP KEY>");
+clientOptions.environment = 'sandbox';
+clientOptions.logLevel = ably.LogLevel.verbose;
+```
+
+##### Rest API
+
+```dart
+ably.Rest rest = await ablyPlugin.createRest(options: clientOptions);
+```
+
+Getting a channel instance
+```dart
+ably.RestChannel channel = rest.channels.get('test');
+```
+
+Publish rest messages
+
+```dart
+//passing both name and data
+await channel.publish(name: "Hello", data: "Ably");
+//passing just name
+await channel.publish(name: "Hello");
+//passing just data
+await channel.publish(data: "Ably");
+//publishing an empty message
+await channel.publish();
+```
+
+##### Realtime API
+
+```dart
+ably.Realtime realtime = await ablyPlugin.createRealtime(options: clientOptions);
```

Comment:
Why does `createRealtime` and `createRest` need to be async?

Reply:
@mattheworiordan As they require to contact platform methods via platform channel which is an async operation.

Reply:
> As they require to contact platform methods via platform channel which is an async operation.

But is that necessary?  
Specifically, if Realtime or REST asynchronously creates an instance of Realtime or REST with iOS or Android, and another method is called immediately afterwards, wouldn't the Dart instance exist synchronously and thus be ready to proxy requests anyway?  If not, does this introduce much complexity to fix?

Reply:
Dart instance will be created only after platform instance is created.

Creating a Rest/Realtime instance on dart side will require a `long` identifier, which will be sent back to dart to store against the dart instance.


#### Here is how it is working now:

Any methods called on dart side will be triggered on the platform side instance  - this instance will be queried using the `long` identifier sent from dart side for each method call.

```dart
createRest ==(async)==> restPlatform(id: 1) ==(async)=> restDart(id: 1)
//await for the above
restInstance = restDart(id: 1)
restDart(id: 1).someMethod(arguments) ==(async)==> restPlatform(id: 1).someMethod(decodedArguments) ==(async)==> responseFromPlatformSide
```

If we don't have that `long` identifier(`id: 1` above) available, we wouldn't know on which platform instance respective method should be invoked.


#### If we created a dart side instance first here is how it might work:

We will need to keep a tab on any methods called immediately after dart side instance is created and trigger platform method call after acquiring the `long` identifier from platforms end.

```dart
createRest ==(async)==> restPlatform(id: 1) ==(async)=> restDart(id: 1)
//don't await for the above
restInstance = restDart(id: null)
restDart.someMethod(arguments) ==(wait until id is acquired)==x
restDart.id = 1;
//then trigger
restDart.someMethod(arguments).__internal__delayed__methodcall__handler.call() ==> restPlatform(id: 1).someMethod(decodedArguments)
```

IMO, this make the dart side code trickier to develop and maintain.

Reply:
Even this should be doable.

Reply:
@mattheworiordan I'd prefer to try and this handled in a separate PR as current PR is purely focused on proving Event handling

@QuintinWillison @paddybyers

Reply:
Moved to handle under [related thread](https://github.com/ably/ably-flutter/pull/6#discussion_r441465807).